contents/docs/product-analytics/group-analytics.mdx

```diff
@@ -24,10 +24,110 @@ import JSSetGroupProperties from "../\_snippets/setting-group-properties.mdx"
 
 <JSCreateGroups />
 
-### Setting and updating group properties
+## How to set group properties
 
 <JSSetGroupProperties />
 
+## How to capture group events
+
+Events are automatically linked to the groups that the user who submitted the event belongs to. Thus you do not add any additional code once you have linked a user to the group. This is the recommended way to capturing group events.
```

Comment:
```suggestion
In the JavaScript web library, groups are automatically linked to the user who triggers an event. Thus you do not add any additional code once you have linked a user to a group. This is the recommended way to capturing group events.
```
This is only true on the frontend?

Reply:
From what I can tell, this isn't the case. See this example:
![Screen Shot 2023-07-07 at 8 29 28 AM](https://github.com/PostHog/posthog.com/assets/34755028/ced40502-b0ae-4669-8ab4-736b992e2095)

user_signed_up event happened first with groups properties, then finished onboarding, but finished onboarding doesn't have group properties.

Reply:
Nope, this isn't true, ever.

There's no concept of a user belonging to a group, so events for a person are never linked to a group. You always have to mention groups property on an event to tell us which group this belongs to.

Just like how every event requires a distinct ID to tell us which user it belongs to.

The difference/confusion here might be this:
1. On the client side, we "save" the group, if any, and add it to all events automatically.
2. On the server side, you explicitly have to add group to all events.

This is similar to how you don't need to add a distinct ID on the client side because we save it, but you do on the server side, since it might not be the same user for the subsequent requests.

Reply:
Thanks both for calling this out!

> You always have to mention groups property on an event to tell us which group this belongs to.

I'll update the docs to reflect this.